frontend/src/pages/_app.tsx

```diff
@@ -105,8 +113,11 @@ const AppContextWrapper: React.FC<ContextProps> = React.memo((props) => {
   );
 });
 
+let extraAppInitialPropsCache: ExtraAppInitialProps;
```

Comment:
This weird trick is needed because custom app unfortunately [doesn't have](https://github.com/vercel/next.js/discussions/10874) `getServerSideProps` yet, and we have to keep initial props somewhere when navigating between pages.